examples/03_connectivity/plot_compare_decomposition.py

```diff
@@ -34,64 +40,99 @@
+from nilearn.image import iter_img
+from nilearn.plotting import plot_stat_map, show
+
+for i, cur_img in enumerate(iter_img(canica_components_img)):
+    plot_stat_map(cur_img, display_mode="z", title="IC %d" % i,
+                  cut_coords=1, colorbar=False)
+
+
+####################################################################
+# Compare CanICA to dictionary learning
+# -------------------------------------------------------------
+# Dictionary learning is a sparsity based decomposition method for extracting
+# spatial maps. It extracts maps that are naturally sparse and usually cleaner
+# than ICA. Here, we will compare networks built with CanICA to networks built
+# with Dictionary Learning.
+#
+#   * Arthur Mensch et al. `Compressed online dictionary learning for fast resting-state fMRI decomposition
+#    <https://hal.archives-ouvertes.fr/hal-01271033/>`_,
+#    ISBI 2016, Lecture Notes in Computer Science
+#
+
+
+###############################################################################
+# Create a dictionary learning estimator
+# ---------------------------------------------------------------
+from nilearn.decomposition import DictLearning
+
+dict_learning = DictLearning(n_components=20,
                              memory="nilearn_cache", memory_level=2,
                              verbose=1,
                              random_state=0,
                              n_epochs=1,
                              mask_strategy='template')
-###############################################################################
-# CanICA
-# ------
-canica = CanICA(n_components=n_components,
-                memory="nilearn_cache", memory_level=2,
-                threshold=3.,
-                n_init=1,
-                verbose=1,
-                mask_strategy='template')
 
-###############################################################################
-# Fit both estimators
-# --------------------
-estimators = [dict_learning, canica]
-names = {dict_learning: 'DictionaryLearning', canica: 'CanICA'}
-components_imgs = []
-
-for estimator in estimators:
-    print('[Example] Learning maps using %s model' % names[estimator])
-    estimator.fit(func_filenames)
-    print('[Example] Saving results')
-    # Grab extracted components umasked back to Nifti image.
-    # Note: For older versions, less than 0.4.1. components_img_
-    # is not implemented. See Note section above for details.
-    components_img = estimator.components_img_
-    components_img.to_filename('%s_resting_state.nii.gz' %
-                               names[estimator])
-    components_imgs.append(components_img)
+print('[Example] Fitting dicitonary learning model')
+dict_learning.fit(func_filenames)
+print('[Example] Saving results')
+# Grab extracted components umasked back to Nifti image.
+# Note: For older versions, less than 0.4.1. components_img_
+# is not implemented. See Note section above for details.
```

Comment:
I would remove the notes about the versions older than 0.4.1. They are quite old, and we don't have to worry about them.